### PR TITLE
[build] Fix version and remove operator-sdk build 

### PR DESCRIPTION
**Pull request by PratikMahajan**
this commit removes the usage of `operator-sdk build`
command to build the operator image.
We are now building the operator binary in dockerfile
similar to what we are doing in dockerfile.ci and
using `make build` which uses `build/build.sh` to set
the version for operator binary. Removed other
references of setting version in `hack/olm.sh`

Also included building the WMCO binary using
`make build` while launching the tests through local
machine.
We use this binary for validating the version
annotation on the operator.
This change is needed as `operator-sdk build`
used to generate the operator binary before building
the container image. Now that we are using
`podman build`, we need to explicitly use
`make build` to generate operator binary for e2e test.
Only e2e tests launched through local machine needs
this change. We use different `wmcoPath` while running
the test through CI based on how we generate the
container image.

Removing `operator-sdk build` dependency is important
because `operator-sdk build` is being depreciated in
operator-sdk version 1.x.

Also, this enables us to trigger automatic builds on
quay

### CODE REVIEW

File: hack/common.sh [hunk 1]
```diff
@@ -50,8 +50,7 @@ build_WMCO() {
       error-exit "OPERATOR_IMAGE not set"
   fi
 
-  $OSDK build "$OPERATOR_IMAGE" --image-builder $CONTAINER_TOOL $noCache \
-    --go-build-args "-ldflags -X=github.com/openshift/windows-machine-config-operator/version.Version=${VERSION}"
+  $CONTAINER_TOOL build . -t "$OPERATOR_IMAGE" -f build/Dockerfile $noCache
```
Comment (aravindhp) [inline hack/common.sh:53]: This is not deprecating the SDK  build but removing it. I am fine with it as long as rest of the team agrees but you need to frame the PR differently.

File: hack/common.sh [hunk 2]
```diff
@@ -132,7 +131,7 @@ get_version() {
   GIT_COMMIT=$(git rev-parse --short HEAD)
   VERSION="${OPERATOR_VERSION}+${GIT_COMMIT}"
 
-  if [ -n "$(git status --porcelain)" ]; then
+  if [ -n "$(git status version tools.go go.mod go.sum vendor build cmd hack pkg --porcelain)" ]; then
```
Comment (aravindhp) [inline hack/common.sh:134]: Missing `Makefile` here.

File: hack/common.sh (hunk 1 above)
Comment (aravindhp) [inline hack/common.sh:53]: Does this mean that `operator-sdk build` will not work now? If that is the case, you also need to update [HACKING.md](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/HACKING.md). If `operator-sdk build` works then you don't need to update the doc.

File: hack/olm.sh [hunk 3]
```diff
@@ -42,7 +42,6 @@ source $WMCO_ROOT/hack/common.sh
 
 cd $WMCO_ROOT
 OSDK=$(get_operator_sdk)
-VERSION=$(get_version)
```
Comment (aravindhp) [inline hack/olm.sh:45]: Why is this change needed? A question like this implies that it warrants a mention in the commit message.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @PratikMahajan. Please address my comments.

File: hack/common.sh (hunk 2 above)
Comment (PratikMahajan) [reply hack/common.sh:134]: Ah!! Thanks for catching !

File: hack/common.sh (hunk 1 above)
Comment (PratikMahajan) [reply hack/common.sh:53]: it should not break `operator-sdk build` command and it will work. But now that we are removing it, i think we should remove it from doc as well.

File: hack/olm.sh (hunk 3 above)
Comment (PratikMahajan) [reply hack/olm.sh:45]: changed the commit message

File: hack/common.sh (hunk 2 above)
Comment (aravindhp) [reply hack/common.sh:134]: I can change the way the operator is built by modifying the `Makefile`. It needs to be present here. Moreover you don't get anything by **not** adding it here.

**Review by aravindhp — CHANGES_REQUESTED** [review]

**Review by aravindhp — APPROVED** [review]
/approve

File: hack/common.sh [hunk 4]
```diff
@@ -132,7 +131,7 @@ get_version() {
   GIT_COMMIT=$(git rev-parse --short HEAD)
   VERSION="${OPERATOR_VERSION}+${GIT_COMMIT}"
 
-  if [ -n "$(git status --porcelain)" ]; then
+  if [ -n "$(git status version tools.go go.mod go.sum vendor Makefile build cmd hack pkg --porcelain)" ]; then
```
Comment (sebsoto) [inline hack/common.sh:134]: I dont understand why we are being so specific here, it will make it a pain to update in the future. Can you call out in the code why this needs to be done?

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: hack/common.sh (hunk 4 above)
Comment (PratikMahajan) [reply hack/common.sh:137]: Added comment in `dockerfile` and `common.sh`

File: hack/common.sh (hunk 1 above)
Comment (aravindhp) [reply hack/common.sh:53]: Just realized you have not updated the doc.

**Comment by PratikMahajan** [comment]
/hold 
until we fix the tests

File: build/Dockerfile.ci [hunk 5]
```diff
@@ -68,6 +54,24 @@ COPY containernetworking-plugins/ .
 ENV CGO_ENABLED=0
 RUN ./build_windows.sh
 
+# Build WMCO
+WORKDIR /build/windows-machine-config-operator
+# Copy files and directories needed to build the WMCO binary
+COPY build build
```
Comment (mansikulkarni96) [inline build/Dockerfile.ci:60]: is this the only folder that is required to be transferred at the end to solve the dirty tag issue, or do we need something else too?

File: build/Dockerfile.ci [hunk 6]
```diff
@@ -14,25 +14,11 @@ RUN curl -L -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp/4.6.4/o
     && tar -xzf openshift-origin-client-tools.tar.gz \
     && rm -rf ./{openshift*,README.md}
 
-# Build WMCO
 # The source here corresponds to the code in the PR and is placed here by the CI infrastructure.
-WORKDIR /build/windows-machine-config-operator/
+RUN mkdir /build/windows-machine-config-operator
```
Comment (aravindhp) [inline build/Dockerfile.ci:18]: This is not needed

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: build/Dockerfile.ci (hunk 5 above)
Comment (PratikMahajan) [reply build/Dockerfile.ci:59]: We need to transfer .gitignore and move building the WMCO operator after building all submodules. That will remove the dirty tag.

**Review by aravindhp — APPROVED** [review]
/approve
